plugins/inputs/varnish/varnish.go

```diff
@@ -0,0 +1,162 @@
+package varnish
+
+// varnish.go
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	kwAll = "all"
+)
+
+// Varnish is used to store configuration values
+type Varnish struct {
+	Stats  []string
+	Binary string
+}
+
+var defaultStats = []string{"MAIN.cache_hit", "MAIN.cache_miss", "MAIN.uptime"}
+var defaultBinary = "/usr/bin/varnishstat"
+
+var varnishSampleConfig = `
+  ## The default location of the varnishstat binary can be overridden with:
+  binary = "/usr/bin/varnishstat"
+
+  ## By default, telegraf gather stats for 3 metric points.
+  ## Setting stats will override the defaults (["%s"]).
+  ## stats may also be set to ["all"], which will collect all stats
+  stats = ["%[1]s"]
```

Comment:
I've used formatted string to avoid repeating the defaultStats list in multiple places. I'll clean it up and update the README and CHANGELOG as per your comments on my other PRs.